tests/test_spag.py

```diff
@@ -514,6 +517,53 @@ def test_spag_template_shortcut(self):
         self.assertEqual(json.loads(out), {"id": "wumbo"})
         self.assertEqual(ret, 0)
 
+    def test_spag_template_default(self):
+        _, err, ret = run_spag('request', 'template/post_thing',
+                               '--with', 'thing_id=mydefaultid')
+        self.assertEqual(err, '')
+        self.assertEqual(ret, 0)
+
+        out, err, ret = run_spag('request', 'template/get_default')
+        self.assertEqual(err, '')
+        self.assertEqual(json.loads(out), {"id": "mydefaultid"})
+        self.assertEqual(ret, 0)
+
+    def test_spag_template_from_default_and_active_environments(self):
```

Comment:
So two things about this, not sure if either of them are because of this PR.

1. If you have headers set in your environment and you use the request
```
(env)~/Desktop/noodles$ spag request get_active_env
{
  "Mini": "barnacle",
  "Thing": "scooby",
  "Wumbo": "mermaid"
}
(env)~/Desktop/noodles$ spag request get_active_env
(env)~/Desktop/noodles$ spag env set -H foo:bar
dir: /Users/Tim/Desktop/noodles/tests/templates
endpoint: http://localhost:5000
headers:
  foo: bar
mini: barnacle
thing: scooby
wumbo: mermaid

(env)~/Desktop/noodles$ spag request get_active_env
{
  "Foo": "bar"
}
(env)~/Desktop/noodles$
```

2. Those environment variables with spaces don't work IRL, just in the test.
```
$ spag env set mini=barnacle boy wumbo=mermaid man thing=scooby doo
Traceback (most recent call last):
  File "/Users/Tim/Desktop/noodles/env/bin/spag", line 9, in <module>
    load_entry_point('spag==0.0.1', 'console_scripts', 'spag')()
  File "/Users/Tim/Desktop/noodles/env/lib/python2.7/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "/Users/Tim/Desktop/noodles/env/lib/python2.7/site-packages/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/Users/Tim/Desktop/noodles/env/lib/python2.7/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/Tim/Desktop/noodles/env/lib/python2.7/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/Tim/Desktop/noodles/env/lib/python2.7/site-packages/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/Tim/Desktop/noodles/env/lib/python2.7/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/Users/Tim/Desktop/noodles/spag/cli.py", line 194, in env_set
    envvars = {key: value for (key, value) in [e.split('=') for e in envvars]}
  File "/Users/Tim/Desktop/noodles/spag/cli.py", line 194, in <dictcomp>
    envvars = {key: value for (key, value) in [e.split('=') for e in envvars]}
ValueError: need more than 1 value to unpack
```